shm/Cargo.hpp

```diff
@@ -0,0 +1,33 @@
+#include <string>
+
+class Cargo {
+public:
+    Cargo() = default;
```

Comment:
I'm not sure how I feel about default constructor here. Since `Ship` is supposed to hold `Cargo` it cannot hold undefined type of cargo. Every object should be defined on creation.